modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/Scala3StepFilter.scala

```diff
@@ -12,18 +12,33 @@ import scala.util.Try
 import java.nio.file.Path
 import scala.util.Success
 import scala.util.Failure
+import java.util.Optional
 import ch.epfl.scala.debugadapter.ScalaVersion
 
 class Scala3StepFilter(
     scalaVersion: ScalaVersion,
     bridge: Any,
-    skipMethod: Method
+    skipMethod: Method,
+    formatMethod: Method
 ) extends ScalaStepFilter(scalaVersion) {
-  override protected def skipScalaMethod(method: jdi.Method): Boolean =
+  override protected def skipScalaMethod(method: com.sun.jdi.Method): Boolean = {
     try skipMethod.invoke(bridge, method).asInstanceOf[Boolean]
     catch {
       case e: InvocationTargetException => throw e.getCause
     }
+  }
+
+  override def formatScala(method: jdi.Method): Option[String] = {
+    try {
+
+      val optionalFormatting = formatMethod.invoke(bridge, method).asInstanceOf[Optional[String]]
+      if (!formatMethod.invoke(bridge, method).asInstanceOf[Optional[String]].isPresent()) None
+      else Some(optionalFormatting.get())
```

Comment:
If you import `scala.jdk.converters.OptionConverters._` you can then do

```suggestion
      formatMethod.invoke(bridge, method).toScala
```